src/python/turicreate/toolkits/activity_classifier/_activity_classifier.py

```diff
@@ -176,6 +176,23 @@ def create(dataset, session_id, target, features=None, prediction_window=100,
             raise _ToolkitError('Unknown advanced parameters: {}'.format(unsupported))
         params.update(kwargs['_advanced_parameters'])
 
+    if isinstance(validation_set, str) and validation_set == 'auto':
```

Comment:
Is this for an auto split? C++ has that functionality.

Reply:
Yes it does, thanks for pointing it out.
The thing is it seems like the implementation might be different?
@nickjong is it better to done it in Python side or C++ for this case :).

Reply:
It is better to use C++ as far as I think. The minimum sessions needed for split are different. In python, it is 100 while in C++ it is 50. I think you would have to modify the tests to accommodate this change.